predicate.go

```diff
@@ -0,0 +1,219 @@
+		var buf bytes.Buffer
+		body := io.TeeReader(r.Body, &buf)
+		var extenderBindingArgs schedulerapi.ExtenderBindingArgs
+
+		if err := json.NewDecoder(body).Decode(&extenderBindingArgs); err != nil {
+			glog.Errorf("json decode error: %v", err)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+		}
+		b := &v1.Binding{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: extenderBindingArgs.PodNamespace,
+				Name:      extenderBindingArgs.PodName,
+				UID:       extenderBindingArgs.PodUID,
+			},
+			Target: v1.ObjectReference{
+				Kind: "Node",
+				Name: extenderBindingArgs.Node,
+			},
+		}
+		err := clientset.CoreV1().Pods(b.Namespace).Bind(b)
+		if err != nil {
+			glog.Errorf("bind failed: %v", err)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func filter(extenderArgs schedulerapi.ExtenderArgs, extenedResourceScheduler *ExtendedResourceScheduler) *schedulerapi.ExtenderFilterResult {
+	pod := extenderArgs.Pod
+	nodes := extenderArgs.Nodes.Items
+
+	canSchedule := make([]v1.Node, 0, len(extenderArgs.Nodes.Items))
+	canNotSchedule := make(map[string]string)
+	// default all node scheduling failed
+	defaultNotSchedule := defaultFailedNodes(nodes)
+
+	result := &schedulerapi.ExtenderFilterResult{
+		Nodes: &v1.NodeList{
+			Items: canSchedule,
+		},
+		FailedNodes: defaultNotSchedule,
+		Error:       "",
+	}
+
+	ercs, err := findExtendedResourceClaims(pod, extenedResourceScheduler)
+	if err != nil {
+		glog.Errorf("find extendedresourceclaims failed: %v", err)
+		result.Error = err.Error()
+		return result
+	}
+	// Find ExtendedResourceClaim by extendedresourceclaim's name
+	for _, erc := range ercs {
+		erList, err := findExtendedResourceList(erc, extenedResourceScheduler)
+		if err != nil {
+			glog.Errorf("erc: %v error: %v", erc, err)
+			result.Error = err.Error()
+			return result
+		}
+
+		extendedResources := &v1alpha1.ExtendedResourceList{
+			Items: make([]v1alpha1.ExtendedResource, 0),
+		}
+		for _, er := range erList.Items {
+			nodeAffinity := er.Spec.NodeAffinity
+			requirements := erc.Spec.MetadataRequirements
+
+			if nodeAffinity == nil ||
+				!mapInMap(requirements.MatchLabels, er.Spec.Properties) ||
+				!labelMatchesLabelSelectorExpressions(requirements.MatchExpressions, er.Spec.Properties) {
+				continue
+			}
+
+			nodeSelectorTerms := nodeAffinity.Required.NodeSelectorTerms
+			for _, node := range nodes {
+				// set er related with erc
+				if nodeMatchesNodeSelectorTerms(&node, nodeSelectorTerms) {
+					erc.Spec.ExtendedResourceNames = append(erc.Spec.ExtendedResourceNames, er.Name)
+					er.Spec.ExtendedResourceClaimName = erc.Name
+					extendedResources.Items = append(extendedResources.Items, er)
+					canSchedule = append(canSchedule, node)
```

Comment:
why do you add this node here ? can you make sure that the ERs in this node can satisfy the reuirements of ERC ?

Reply:
we find this node through ER affinity, why ER can not satisfy the requirements of ERC, if not, it's a binding error between ER and node?

Reply:
Why do you want to find node through ER affinity ?
And one node which we can schedule pod to must have enough ERs for erc requirements.

Reply:
yes, but I don't know how to find and filter node except though ER affinity